src/content/docs/extras.mdx

```diff
@@ -320,6 +320,213 @@ str_detect("Hello world", "ello")
+```r
+dat = data.table::fread("https://raw.githubusercontent.com/Rdatatable/data.table/master/vignettes/flights14.csv")
+
+dir.create("flights14")
+arrow::write_dataset(
+  dat, 
+  path = "flights14",
+  partitioning = c("origin") # <-- NB: partition your data by the main grouping vars for efficiency
+)
+```
+
+The `write_dataset` and `open_dataset` functions are useful for very large datasets and for efficiently "pre-grouping" the data (e.g. if you know you're going to `group_by(origin)` a lot, then saving the dataset this way is efficient). If you want a single file rather than a folder system, you can read/write to the `.parquet` file format. The parquet version will be much smaller in size (in this case, the `.csv` is 9.6MB versus the `.parquet` at 1.4MB) and load/save faster.
+
+```r
+arrow::write_parquet(dat, "flights14.parquet")
+```
+
+We can load 
+```r
+dat = arrow::open_dataset("flights14/")
+# or, 
+# dat = arrow::read_parquet("flights14.parquet")
+
+# View arrow database scheme
+dat
+```
+```text
+FileSystemDataset with 3 Parquet files
+year: int32
+month: int32
+day: int32
+dep_delay: int32
+arr_delay: int32
+carrier: string
+dest: string
+air_time: int32
+distance: int32
+hour: int32
+origin: string
+
+See $metadata for additional Schema metadata
+```
+
+#### Data wrangling with arrow
+
+With our dataset loaded, we can begin to do standard data modifying using `dplyr`. 
+
+```r
+sum = dat |> 
+  filter(origin == "JFK") |>
+  group_by(month) |>
+  summarize(
+    mean_dep_delay = mean(dep_delay, na.rm = TRUE), 
+    n_flights = n()
+  )
+```
+
+If you type sum into the console, the output will not look like a data.frame. Instead, you'll see that a `query` object has been created:
+
+```
+sum
+```
+```text
+FileSystemDataset (query)
+month: int32
+mean_dep_delay: double
+n_flights: int64
+
+See $.data for the source Arrow object
+```
+
+Instead of loading the output, `arrow` has "prepared" a set of commands that will need to be triggered with the `collect()` function. Note that once you `collect()`, the result returned is a standard R data.frame, so do so carefully.
+
+```r
+sum |> 
+  collect()
+```
+```text
+# A tibble: 10 × 3
+   month mean_dep_delay n_flights
+   <int>          <dbl>     <int>
+ 1     1          27.0       7105
+ 2     2          19.2       6779
+ 3     3           7.17      8253
+ 4    10           6.97      8605
+ 5     4           7.00      8070
+ 6     5          10.9       8305
+ 7     6           9.81      8422
+ 8     7          17.6       8730
+ 9     8           8.66      8986
+10     9           3.76      8228
+```
+
+Under the hood, arrow uses the computation engine, [acero](https://arrow.apache.org/docs/cpp/streaming_execution.html). This computation engine is what allows arrow to do really efficient operations on large datasets. However, it is not yet as fully featured as duckdb's computation engine is. We will discuss more below.
```

Comment:
Suggest slight rewording: "However, it is not as fast or fully featured as DuckDB's computation engine."